GoogleForCreators/web-stories-wp
# Floating Menu Return Focus to Element

## Context
Part of canvas keyboard nav work.


## Summary
Makes focus groups a little more stable, to where if you click into them, they behave the same as if you keyboard nav into them keybinding wise.

Also fixes exiting in/out of focus groups properly between floating menu -> element -> canvas focus wrapper.


## Relevant Technical Choices
Decided to have all focus groups rendered all the time. This works, because the only way to get into them is to manually focus (via click) or call one of the `EditLayerFocusManager` actions. This simplifies things and adds consistent behavior.

Also got rid of the concept of an `active focus group`. They're all always active now, and you can just navigate in/out of them with mouse or keybindings.

Added group specific cleanup functions since they're relatively static for the lifetime of the components. This fixes the focus order:
- in floating menu focus group
- press esc
- in element selection focus group
- press esc
- focused on canvas focus wrapper


## To-do
- [x] add some test coverage of new functionality


## User-facing changes
Improved consistency of interactions between mouse & keyboard nav with canvas focus groups


## Testing Instructions






This PR can be tested by following these steps:

1. focus the sidepanel
2. click on an element
3 press `ctrl+alt+p` to keyboard nav into the floating menu
4. press `esc`
5 see that selected element has focus again
6. press `tab` or `shift/tab` a bunch and see that you're only navigating between elements
7. press `esc`
8. See that your focused on the canvas wrapper


## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #11003
